citest/json_predicate/logic_predicate.py

```diff
@@ -112,6 +118,9 @@ def __init__(self, pred):
   def __str__(self):
     return 'NOT ({0})'.format(self.__pred)
```

Comment:
Isnt {0} well understood as string conversion?.
I'm going to change the __repr__ to produce structured JSON instead of
adhoc strings so that it can be reformatted to be readable. The strings get
very nasty as they compose to higher and higher levels. But am not going to
change it until later because of other priorities.

On Thu, Apr 7, 2016 at 10:45 AM, Lars Wander <notifications@github.com>
wrote:

> In citest/json_predicate/logic_predicate.py
> <https://github.com/google/citest/pull/137#discussion_r58883556>:
>
> > @@ -112,6 +118,9 @@ def __init__(self, pred):
> >    def __str__(self):
> >      return 'NOT ({0})'.format(self.__pred)
>
> You can force the __str__ representation by changing this to {0!s}. I
> think that's the default, but it might be clearer since you're forcing the
> __repr__ representation below.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/google/citest/pull/137/files/a82ce9aa13cfe030abd936cc0de1ee79176f2ab2#r58883556>
>